### PR TITLE
fix #181 : custom title in MappingEditor

### LINKED ISSUE #181: custom title in a MappingEditor (session viewer) is ignored
It is overwritten by the selected array name, instead of being prepended to it.

### PR DESCRIPTION
**Pull request by alixdamman**
displayed title = 'associated file - name : axes_info - custom part' if LArray object, else 'name - custom part'

I will include #256 after your review on current commit (fix #181)

### CODE REVIEW

File: larray/viewer.py [hunk 1]
```diff
@@ -1983,11 +1983,13 @@ def setup_and_check(self, data, title='', readonly=False, minvalue=None, maxvalu
...
-            title = _("Session viewer") if readonly else _("Session editor")
+        if title:
+            self.title = title
+        else:
+            self.title = _("Session viewer") if readonly else _("Session editor")
         if readonly:
-            title += ' (' + _('read only') + ')'
-        self.setWindowTitle(title)
+            self.title += ' (' + _('read only') + ')'
+        self.setWindowTitle(self.title)
```
Comment (gdementen) [inline larray/viewer.py]: in this kind of case, I prefer to use simply "title" and set self.title at the end.

File: larray/viewer.py [hunk 2]
```diff
@@ -2292,14 +2294,24 @@ def on_item_changed(self, curr, prev):
...
             axes = array.axes
             axes_info = ' x '.join("%s (%d)" % (display_name, len(axis))
                                    for display_name, axis
                                    in zip(axes.display_names, axes))
-            title = (title + ': ' + axes_info) if title else axes_info
-        self.setWindowTitle(title)
+            title += [(name + ': ' + axes_info) if name else axes_info]
+        elif name:
+            title = [name]
+        title += [self.title]
```
Comment (gdementen) [inline larray/viewer.py]: we might want to move this whole "title construction" to its own method

**Comment by gdementen** [comment]
as I said in #257, I would postpone fixing #256 to 0.23 (unless you have done it already)

**Comment by alixdamman** [comment]
OK to postpone #256 to 0.23

**Comment by alixdamman** [comment]
rebase done + updated my commit "fix 181..."

File: larray/viewer.py [hunk 3]
```diff
@@ -2295,16 +2296,34 @@ def on_item_changed(self, curr, prev):
             else:
                 self.eval_box.setText(expr)
 
-    # TODO: rename to set_current_array
-    def set_widget_array(self, array, title):
+    def update_title(self):
+        current_item = self._listwidget.currentItem()
+        name = str(current_item.text()) if current_item else ''
+        array = self.data[name] if name else None
```
Comment (gdementen) [inline larray/viewer.py]: does this work for `<expr>`? (computed in the console)?

Comment (alixdamman) [inline larray/viewer.py]: If qtconsole_available, yes.

**Comment by gdementen** [comment]
LGTM then

File: larray/viewer.py (hunk 3 above)
Comment (gdementen) [inline larray/viewer.py]: well... It does not... I will fix this.

Comment (alixdamman) [inline larray/viewer.py]: I tried with something like 
[1] arr4 = arr4.reshape(...)
which updated the title correctly. 
I guess I misunderstood when view_expr() is called.

Comment (gdementen) [inline larray/viewer.py]: it is called if you try something like:
`arr4.reshape(...)` (without assigning it to anything)

**Pull request merged**